docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -73,19 +129,53 @@ the ``copy`` module::
     ...
     '''
 
-Building & Testing
+Formatting options
 ''''''''''''''''''
+These formatting functions are ``U()``, ``M()``, ``I()``, and ``C()``
+for URL, link to help for another module, `italic`, and ``constant-width`` respectively. It is suggested
+to use ``C()`` for file and option names, and ``I()`` when referencing
+parameters; module names should be specified as ``M(module)`` to create a
+link to the online documentation for that module.
+
 
-Put your completed module file into the 'library' directory and then
+Example usage::
+
+    Or if not set the environment variable C(ACME_PASSWORD) will be used.
+    ...
+    Required if I(state=present)
```

Comment:
I've added some comments jtyr's PR https://github.com/ansible/ansible/pull/20931 and we can have that discussion there.